### PR TITLE
Define 'tick'

### PR DESCRIPTION
**Pull request by kelunik**

### CODE REVIEW

File: src/Loop/Driver.php [hunk 1]
```diff
@@ -15,7 +15,14 @@
     private $registry = [];
 
     /**
-     * Start the event loop.
+     * Run the event loop.
+     *
+     * One iteration of the loop is called one "tick". A tick covers the following steps:
+     *
+     *  1. Activate watchers created / enabled in the last tick / before `run`.
```
Comment (bwoebi) [inline src/Loop/Driver.php]: With \`run\` you are referencing the method? Please suffix `()` then to make this more obvious.

File: src/Loop/Driver.php [hunk 2]
```diff
@@ -15,7 +15,14 @@
...
+     *  2. Execute all enabled defer watchers.
+     *  3. Execute all due timers, each timer only once per tick.
+     *  4. Execute actionable stream watchers, each only once per tick.
```
Comment (bwoebi) [inline src/Loop/Driver.php]: Uh, there is no order guarantee between timers and stream watchers. Merge these two into a single step.

File: src/Loop/Driver.php (hunk 1 above)
Comment (kelunik) [reply src/Loop/Driver.php]: Fixed.

File: src/Loop/Driver.php (hunk 2 above)
Comment (kelunik) [reply src/Loop/Driver.php]: Fixed, once per tick should be fine?

Comment (bwoebi) [reply src/Loop/Driver.php]: yes, that's fine.

**Review by bwoebi — APPROVED** [review]

File: src/Loop/Driver.php [hunk 3]
```diff
@@ -15,7 +15,13 @@
...
 
     /**
-     * Start the event loop.
+     * Run the event loop.
+     *
+     * One iteration of the loop is called one "tick". A tick covers the following steps:
+     *
+     *  1. Activate watchers created / enabled in the last tick / before `run()`.
+     *  2. Execute all enabled defer watchers.
+     *  3. Execute all due timer and actionable stream callbacks, each only once per tick.
```
Comment (trowski) [inline src/Loop/Driver.php]: Can signals be executed at any time or should they be included in this step? All current implementations execute pending signals only in this step, but should that be part of the spec?

**Review by trowski — CHANGES_REQUESTED** [review]

File: src/Loop/Driver.php (hunk 3 above)
Comment (bwoebi) [reply src/Loop/Driver.php]: Yes, they should. Missed that.

Comment (kelunik) [reply src/Loop/Driver.php]: Changed.

**Review by trowski — APPROVED** [review]

**Pull request merged**